QgisModelBaker/gui/dataset_manager.py

```diff
@@ -0,0 +1,258 @@
+        TID = Qt.UserRole + 1
+        DATASETNAME = Qt.UserRole + 2
+
+        def __int__(self):
+            return self.value
+
+    def __init__(self):
+        super().__init__()
+
+    def flags(self, index):
+        return Qt.ItemIsSelectable | Qt.ItemIsEnabled
+
+    def reload_datasets(self, db_connector):
+        datasets_info = db_connector.get_datasets_info()
+        self.beginResetModel()
+        self.clear()
+        for record in datasets_info:
+            item = QStandardItem()
+            item.setData(record['datasetname'], int(Qt.DisplayRole))
+            item.setData(record['datasetname'], int(DatasetSourceModel.Roles.DATASETNAME))
+            item.setData(record['t_id'], int(DatasetSourceModel.Roles.TID))
+            self.appendRow(item)
+        self.endResetModel()
+
+class DatasetManagerDialog(QDialog, DIALOG_UI):
+
+    def __init__(self, parent=None):
+
+        QDialog.__init__(self, parent)
+        self.setupUi(self)
+        self.buttonBox.accepted.connect(self.accepted)
+        self.buttonBox.rejected.connect(self.rejected)
+
+        self.type_combo_box.clear()
+        self._lst_panel = dict()
+        self.db_simple_factory = DbSimpleFactory()
+
+        for db_id in self.db_simple_factory.get_db_list(False):
+            self.type_combo_box.addItem(displayDbIliMode[db_id], db_id)
+            db_factory = self.db_simple_factory.create_factory(db_id)
+            item_panel = db_factory.get_config_panel(
+                self, DbActionType.EXPORT)
+            self._lst_panel[db_id] = item_panel
+            self.db_layout.addWidget(item_panel)
+
+        self.type_combo_box.currentIndexChanged.connect(self.type_changed)
+
+        self.dataset_model = DatasetSourceModel()
+        self.dataset_tableview.horizontalHeader().setSectionResizeMode(QHeaderView.Stretch)
+        self.dataset_tableview.horizontalHeader().hide()
+        self.dataset_tableview.verticalHeader().hide()
+        self.dataset_tableview.setSelectionMode(QTableView.SingleSelection)
+        self.dataset_tableview.setModel(self.dataset_model)
+
+        self.restore_configuration()
+
+        #refresh the models on changing values but avoid massive db connects by timer
+        self.refreshTimer = QTimer()
+        self.refreshTimer.setSingleShot(True)
+        self.refreshTimer.timeout.connect(self.refresh_datasets)
+
+        for key, value in self._lst_panel.items():
+            value.notify_fields_modified.connect(self.request_for_refresh_datasets)
+
+        self.reload_datasets(self.updated_configuration())
+
+        self.add_button.clicked.connect(self.add_dataset)
+        self.edit_button.clicked.connect(self.edit_dataset)
+        self.create_baskets_button.clicked.connect(self.create_baskets)
+        self.dataset_tableview.selectionModel().selectionChanged.connect( lambda: self.enable_dataset_handling(True))
+
+    def valid_selection(self):
+        return bool(len(self.dataset_tableview.selectedIndexes()))
```

Comment:
```suggestion
    """
    Returns if at least one dataset is selected
    """
        return bool(self.dataset_tableview.selectedIndexes())
```

Should be enough

Reply:
Ok. It doesn't even need the `bool` - but I'm not sure what is the best practice here. Considering "explicit is better than implicit" and "simple is better than complex" I like the version you suggested.